win32/src/file_win32.c

```diff
@@ -60,4 +41,472 @@ static void on_file_io_complete_win32(
+            handle, source, size, position, user_callback, user_context);
+        result = FILE_WRITE_ASYNC_INVALID_ARGS;
+    }
+    else
+    {
+        bool callback_will_be_called = false;
+        /*Codes_SRS_FILE_WIN32_43_018: [ file_write_async shall allocate a context to store the allocated OVERLAPPED struct, handle, size, user_callback and user_context. ]*/
+        FILE_WIN32_IO* io = malloc(sizeof(FILE_WIN32_IO));
+        if (io == NULL)
+        {
+            /*Codes_SRS_FILE_43_015: [ If there are any failures, file_write_async shall fail and return FILE_WRITE_ASYNC_ERROR. ]*/
+            /*Codes_SRS_FILE_WIN32_43_057: [ If there are any other failures, file_write_async shall fail and return FILE_WRITE_ASYNC_ERROR.]*/
+            LogError("failure in malloc");
+            result = FILE_WRITE_ASYNC_ERROR;
+        }
+        else
+        {
+            io->handle = handle;
+            /*Codes_SRS_FILE_WIN32_43_020: [ file_write_async shall allocate an OVERLAPPED struct and populate it with the created event and position. ]*/
+            (void)memset(&(io->ov), 0, sizeof(OVERLAPPED));
+            /*Codes_SRS_FILE_WIN32_43_054: [ file_write_async shall create an event by calling CreateEvent.]*/
+            io->ov.hEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+            if (io->ov.hEvent == NULL)
+            {
+                /*Codes_SRS_FILE_43_015: [ If there are any failures, file_write_async shall fail and return FILE_WRITE_ASYNC_ERROR. ]*/
+                /*Codes_SRS_FILE_WIN32_43_057: [ If there are any other failures, file_write_async shall fail and return FILE_WRITE_ASYNC_ERROR.]*/
+                LogError("failure in CreateEvent");
+                result = FILE_WRITE_ASYNC_ERROR;
+            }
+            else
+            {
+                io->ov.Offset = position & 0xFFFFFFFFULL;
+                io->ov.OffsetHigh = position >> 32;
+
+                io->user_callback = user_callback;
+                io->user_context = user_context;
+
+                io->size = size;
+
+                /*Codes_SRS_FILE_WIN32_43_017: [ file_write_async shall call StartThreadpoolIo.]*/
+                StartThreadpoolIo(handle->ptp_io);
+
+                /*Codes_SRS_FILE_43_014: [ file_write_async shall enqueue a write request to write source's content to the position offset in the file. ]*/
+                /*Codes_SRS_FILE_43_041: [ If position + size is greater than the size of the file and the call to write is succesfull, file_write_async shall grow the file to accomodate the write. ]*/
+                /*Codes_SRS_FILE_WIN32_43_021: [ file_write_async shall call WriteFile with handle, source, size and the allocated OVERLAPPED struct.]*/
+                if (WriteFile(handle->h_file, source, size, NULL, &(io->ov)) == FALSE)
+                {
+                    if (GetLastError() == ERROR_IO_PENDING)

```

Comment:
>GetLastError [](start = 24, length = 12)

Store this in a variable because you should also print it out in the failure branch